src/test/scala/com/yotpo/metorikku/code/steps/test/SelectiveMergeTests.scala

```diff
@@ -0,0 +1,170 @@
+    val sortedExpectedDf = df2.orderBy("employee_name").select("employee_name", sortedSchemeArr:_*)
+    val equals = sortedMergedDf.except(sortedExpectedDf).isEmpty
+
+    if (equals != isEqual) {
+      if (!equals) {
+        log.error("Actual and expected differ:")
+        log.error("Actual:")
+        sortedMergedDf.show()
+        log.error("Expected:")
+        sortedExpectedDf.show()
+        assert(false)
+      }
+      else {
+        log.error("Actual and expected are equal (but expected to differ)")
+        assert(false)
+      }
+    }
+  }
+
+  test("Selective merge") {
+    val employeeData1 = Seq(
+      ("James", 1, 11, 111, 1111),
+      ("Maria", 2, 22, 222, 2222)
+    )
+
+    val employeeData2 = Seq(
+      ("James", 1, 33, 333),
+      ("Jen",   4, 44, 444),
+      ("Jeff",  5, 55, 555)
+    )
+
+    val simpleDataExpectedAfterMerge = Seq(
+      ("James", new Integer(1) /* Salary */, new Integer(33) /* age */, new Integer(111) /* fake */,
+        new Integer(1111) /* fake2 */, new Integer(333) /* bonus */),
+      ("Maria", new Integer(2) /* Salary */, new Integer(22) /* age */, new Integer(222) /* fake */,
+        new Integer(2222) /* fake2 */, null.asInstanceOf[Integer] /* bonus */),
+      ("Jen", new Integer(4) /* Salary */, new Integer(44) /* age */, null.asInstanceOf[Integer] /* fake */,
+        null.asInstanceOf[Integer] /* fake2 */, new Integer(444) /* bonus */),
+      ("Jeff", new Integer(5) /* Salary */, new Integer(55) /* age */, null.asInstanceOf[Integer] /* fake */,
+        null.asInstanceOf[Integer] /* fake2 */, new Integer(555) /* bonus */)
+    )
+
+    val simpleDataNotExpectedAfterMerge = Seq(
+      ("James", new Integer(10) /* Salary */, new Integer(33) /* age */, new Integer(111) /* fake */,
+        new Integer(1111) /* fake2 */, new Integer(333) /* bonus */),
+      ("Maria", new Integer(20) /* Salary */, new Integer(22) /* age */, new Integer(222) /* fake */,
+        new Integer(2222) /* fake2 */, null.asInstanceOf[Integer] /* bonus */),
+      ("Jen", new Integer(40) /* Salary */, new Integer(44) /* age */, null.asInstanceOf[Integer] /* fake */,
+        null.asInstanceOf[Integer] /* fake2 */, new Integer(444) /* bonus */),
+      ("Jeff", new Integer(50) /* Salary */, new Integer(55) /* age */, null.asInstanceOf[Integer] /* fake */,
+        null.asInstanceOf[Integer] /* fake2 */, new Integer(555) /* bonus */)
+    )
+
+    val sparkSession = SparkSession.builder.appName("test").getOrCreate()
+    val sqlContext= new SQLContext(sparkSession.sparkContext)
+    import sqlContext.implicits._
+    val df1 = employeeData1.toDF("employee_name", "salary", "age", "fake", "fake2")
+    val df2 = employeeData2.toDF("employee_name", "salary", "age", "bonus")
+    val expectedDf = simpleDataExpectedAfterMerge.toDF("employee_name", "salary", "age", "fake", "fake2", "bonus")
+    val notExpectedDf = simpleDataNotExpectedAfterMerge.toDF("employee_name", "salary", "age", "fake", "fake2", "bonus")
```

Comment:
I don't really see why we need to test `notExpectedDf`

Reply:
Added it because it actually happened to me during development (that is, I played with the comparison at the end of the tests and it passed for every 2 df inputs- passing all tests when it shouldn't have), I added the "not expected" test as a safeguard from that.